float64 rho_projected
float64 theta
int64 class_id